src/background_jobs.rs

```diff
@@ -108,24 +110,45 @@ impl Job {
     ) -> Result<(), EnqueueError> {
         use crate::schema::background_jobs::dsl::*;
 
+        let find_similar_jobs_query = |job: &Job| {
+            let query = background_jobs
+                .select(id)
+                .filter(job_type.eq(job.as_type_str()))
+                .filter(data.eq(job.to_value()?))
+                .filter(priority.eq(PRIORITY_SYNC_TO_INDEX))
```

Comment:
The priority clause isn't in the hand written queries in the PR description — I assume that's an intentional change when you were converting this to the Diesel DSL syntax?

Reply:
yeah, I figured if the `priority` was different there might be a reason for duplicating the job 🤷‍♂️